### DISCUSSION TITLE
How to get "oneOf" in OpenAPI output?

### DISCUSSION THREAD

**Discussion by bstro** [asteasolutions/zod-to-openapi#227] [Q&A]
With the following Zod schema,
```
export const example = z.union([z.object({foo: z.object({…})}), z.object({bar: z.object({…})})])
```
I'm hoping to achieve the following OpenAPI output:
```
"oneOf": [
  {
    "type": "object",
    "properties": {
      "foo": {
        "type": "object",
        "properties": {…}        
      }
    }
  }, {
    "type": "object",
    "properties": {
      "bar": {
        "type": "object",
        "properties": {…}
      }
    }
  }
]
```
I can only seem to get `zod-to-openapi` to render to `anyOf`, but my API has a strict requirement that only one of the objects is passed.

Can anyone offer any pointers??

**Comment by florianmartens** [comment]
Pushing this up!

**Comment by AGalabov** [comment]
I've opened up a PR for this that would eventually be available with a beta (and then an officail) v8 version soo. See the PR here: https://github.com/asteasolutions/zod-to-openapi/pull/308

**Comment by bombillazo** [comment]
Hey, Im interested in the same question, using `union` results in the spec using an any of instead of one of

**Comment by AGalabov** [comment]
This has now been released as an official [v8.0.0 Release](https://github.com/asteasolutions/zod-to-openapi/releases/tag/v8.0.0) 🚀
